vllm/executor/openvino_executor.py

```diff
@@ -0,0 +1,539 @@
+        # which are more optimal for plugins representations
+        return (
+            self.num_heads,
+            self.head_size,
+            self.block_size,
+        )
+
+    def allocate_gpu_cache(self) -> List[OpenVINOKVCache]:
+        if self.num_gpu_blocks == 0:
+            return None
+        gpu_cache: List[OpenVINOKVCache] = []
+        key_block_shape = (self.num_gpu_blocks, *self.get_key_block_shape())
+        value_block_shape = (self.num_gpu_blocks, *self.get_value_block_shape())
+        remote_context = ov.Core().get_default_context("GPU")
+        for _ in range(self.num_layers):
+            key_blocks = remote_context.create_tensor(self.cache_dtype, key_block_shape)
+            value_blocks = remote_context.create_tensor(self.cache_dtype, value_block_shape)
+            gpu_cache.append((key_blocks, value_blocks))
+        return gpu_cache
+
+    def allocate_cpu_cache(self) -> List[OpenVINOKVCache]:
+        cpu_cache: List[OpenVINOKVCache] = []
+        key_block_shape = (self.num_cpu_blocks, *self.get_key_block_shape())
+        value_block_shape = (self.num_cpu_blocks, *self.get_value_block_shape())
+        for _ in range(self.num_layers):
+            key_blocks = ov.Tensor(self.cache_dtype, key_block_shape)
+            value_blocks = ov.Tensor(self.cache_dtype, value_block_shape)
+            cpu_cache.append((key_blocks, value_blocks))
+        return cpu_cache
+
+    def _swap(
+        self,
+        src: List[OpenVINOKVCache],
+        dst: List[OpenVINOKVCache],
+        src_to_dst: Dict[int, int],
+    ) -> None:
+        # TODO: ilavreno: implement cache sync via OpenVINO RemoteContext API host <-> device
+        assert False
+
+    def swap_in(self, src_to_dst: Dict[int, int]) -> None:
+        self._swap(self.cpu_cache, self.gpu_cache, src_to_dst)
+
+    def swap_out(self, src_to_dst: Dict[int, int]) -> None:
+        self._swap(self.gpu_cache, self.cpu_cache, src_to_dst)
+
+    def copy(self, src_to_dsts: Dict[int, List[int]]) -> None:
+        key_caches = [key_cache for key_cache, _ in self.gpu_cache]
+        value_caches = [value_cache for _, value_cache in self.gpu_cache]
+        # TODO: ilavreno: implement cache sync via OpenVINO RemoteContext API device -> device copies
+        # cache_ops.copy_blocks(key_caches, value_caches, src_to_dsts)
+
+    @staticmethod
+    def get_cache_dtype(
+        cache_dtype: str,
+        model_config: ModelConfig,
+        device_config: DeviceConfig) -> ov.Type:
+        # TODO: ilavreno: select kv_cache data type for OpenVINO devices
+        # probably, we need to force OpenVINO kv cache data types per device and assert
+        # if user specified a different value
+        if cache_dtype == "auto":
+            cache_dtype = model_config.dtype
+        else:
+            cache_dtype = STR_DTYPE_TO_TORCH_DTYPE[cache_dtype]
+        ov_cache_dtype = TORCH_DTYPE_TO_OPENVINO_DTYPE[cache_dtype]
```

Comment:
@vladimir-paramuzov @dmitry-gorokhov
here plugins can change precision to store KV caches if required